### DISCUSSION TITLE
To support custom cert CN, O, OU, and expiration date

### DISCUSSION THREAD

**Discussion by m2acgi** [FiloSottile/mkcert#473] [Ideas]
Need to support custom cert CN, O, OU, and expiration date

**Comment by DoraTheExploder** [comment]
I've discovered that if you use an external tool to generate a Certificate Signing Request (.csr) such as [DigiCert CSR Tool](https://www.digicert.com/easy-csr/openssl.htm) (or do it manually with openssl), you can customize the relevant fields and then pass it as a sort of configuration file to mkcert with `$ mkcert -csr <domain>`.

I have not tested using a CSR for initial Root CA installation, but `$ mkcert --help` implies that it should be possible.

**Comment by jet10000** [comment]
https://letsencrypt.org/docs/certificates-for-localhost/
